static/js/message_flags.js

```diff
@@ -122,6 +122,27 @@ export function unstar_all_messages() {
 }
 
 export function unstar_all_messages_in_topic(stream_id, topic) {
-    const starred_message_ids = starred_messages.get_starred_message_ids_in_topic(stream_id, topic);
-    send_flag_update_for_messages(starred_message_ids, "starred", "remove");
+    const data = {
+        anchor: "newest",
+        // We assume that the user does not have more than 100 starred
+        // messages in this topic.
+        num_before: 100,
+        num_after: 0,
```

Comment:
Is this assumption safe to make? Or should I increase the count (or fetched these messages in a loop?)